docs/docs/troubleshooting-common-errors.md

```diff
@@ -0,0 +1,187 @@
+# and place them in the node_modules folder
+npm install
+```
+
+The version of Node.js that's used to install sharp needs to match the versioin of Node.js that is run, so clearing `node_modules` and reinstalling often resolves the problem.
+
+### Incompatible library version: sharp.node requires version X or later, but Y provides version Z
+
+The error `Incompatible library version: sharp.node requires version X or later, but Y provides version Z` means that there are multiple incompatible versions of the `sharp` package installed in `node_modules`. Your error may look something like this:
+
+```
+Something went wrong installing the "sharp" module
+dlopen(/Users/you/gatsby-site/node_modules/sharp/build/Release/sharp.node, 1): Library not loaded: @rpath/libglib-2.0.dylib
+  Referenced from: /Users/you/gatsby-site/node_modules/sharp/build/Release/sharp.node
+  Reason: Incompatible library version: sharp.node requires version 6001.0.0 or later, but libglib-2.0.dylib provides version 5801.0.0
+```
+
+To fix this, you'll need to update all Gatsby plugins in the current project that depend on the `sharp` package. Here's a list of official plugins that you might need to update in case your project uses them:
+
+- `gatsby-plugin-sharp`
+- `gatsby-plugin-manifest`
+- `gatsby-remark-images-contentful`
+- `gatsby-source-contentful`
+- `gatsby-transformer-sharp`
+- `gatsby-transformer-sqip`
+
+To update these packages, run:
+
+```sh
+npm install gatsby-plugin-sharp gatsby-plugin-manifest gatsby-remark-images-contentful gatsby-source-contentful gatsby-transformer-sharp gatsby-transformer-sqip
+```
+
+If updating these doesn't fix the issue, your project probably uses other plugins from the community that depend on a different version of `sharp`. Try running `npm list sharp` or `yarn why sharp` to see all packages in the current project that use `sharp` and try updating them as well.
+
+(using a variable in a static query)
+
+## Errors building and deploying
+
+The process of building your site [varies slightly from the development process](/docs/overview-of-the-gatsby-build-process/). Some errors can arise when you build your site if
+
+For more information on common problems while building your site, refer to the [Debugging HTML Builds](/docs/debugging-html-builds/) guide.
+
+### Build problems from Field 'browser' doesn't contain a valid alias configuration
+
+If you are seeing an error like:
+
+```shell
+Module not found: Error: Can't resolve `../..SomeFile.svg`
+  Field 'browser' doesn't contain a valid alias configuration
+  ...
+```
+
+The build is failing to find the file at `../..SomeFile.svg`. This could be frustrating if your site works when you when run it locally with `gatsby develop`, and even works when you run `gatsby build` and `gatsby serve` locally. A likely problem is that the operating system you are running locally is different than the one where your site is deployed where it's likely running some distribution and flavor of Linux.
```

Comment:
```suggestion
The build is failing to find the file at `../..SomeFile.svg`. This can be frustrating if your site works when you when run it locally with `gatsby develop`, and even works when you run `gatsby build` and `gatsby serve` locally. A likely problem is that the operating system you are running locally is different than the one where your site is deployed. Oftentimes your deployment target is running some distribution and flavor of Linux.
```